packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -107,15 +123,34 @@ class ConfigChainBuilder {
 
     this.resolvedConfigs.push(loc);
 
-    const content = fs.readFileSync(loc, "utf8");
     let options;
+    if (path.extname(loc) === ".js") {
+      try {
+        options = require(loc);
```

Comment:
I think that's a fair point about `.default` and I totally agree for `.babelrc.js`, it is unlikely that it will be compiled so the chance of it having `__esModule` is pretty low.

That said, it still feels most consistent to do that, and the other case is that right now someone can do

```
{
    extends: "./foo.js",
}
```
and `foo.js` will execute, and since it's arbitrary JS, it's possible that it could be a node module with some collected config or something, and that could be transpiled before publish and have `__esModule`.